web/src/components/TimeSliderInput.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+import { sortBy } from 'lodash';
+
+const COLORS = {
+  DAY: '#EFEFEF',
+  NIGHT: 'rgba(0, 0, 0, 0.15)',
+};
+
+const createGradient = sets => sortBy(sets, 'end')
+  .map(
+    ({ start, end }) => `${COLORS.DAY} ${start}%, ${COLORS.NIGHT} ${start}%, ${COLORS.NIGHT} ${end}%, ${COLORS.DAY} ${end}%`,
+  )
+  .join(',\n');
+
+export const StyledInput = styled.input`
+  --time-gradient: linear-gradient(
+    90deg,
+    ${props => createGradient(props.nightSets)}
```

Comment:
not sure to understand what sets are in the context of `nightSets`.

Not sure if it's worth changing appellation, but just to let you know that the name does not obviously explicits what it refers to

Reply:
I have now changed it to `nightTimeSets` to be consistent - but yeah, the naming isn't great.. Any suggestions?

Reply:
already a bit better ;)